howso.amlg

```diff
@@ -769,34 +774,68 @@
 					"labels"
 					(map
 						(lambda
-							(assoc
-								"description" (current_value 1)
-								"parameters"
-									;when debugging, comments are injected to all lines, these must be removed
-									;some parameters may have comments other than the type hint as well
-									(map
-										(lambda
-											;(current_value) is a tuple of [comments of parameter, default value]
-											(append
-												;regex to just capture the {} information
-												(or (parse (substr (first (current_value)) "\\{.+\\}")) (assoc) )
-												(assoc
-													default (last (current_value 1))
-													optional (= (last (current_value 1)) (null))
-													description
-														;replace the typing assoc with an empty string, only get description
-														(substr
-															(first (current_value 1))
-															"\\{.+\\}\\r\\n"
-															(null)
-															""
-														)
+							(let
+								(assoc
+									label_flags_map (parse (substr (current_value 1) "\\{.+\\}"))
+								)
+
+								(assoc
+									;idempotent is true by default
+									"idempotent"
+										(if (contains_index label_flags_map "idempotent")
+											(get label_flags_map "idempotent")
+
+											(true)
+										)
+									;long_running is false by default
+									"long_running"
+										(if (contains_index label_flags_map "long_running")
```

Comment:
for consistency, this is fine. though you could just do `(or (get label_flags_map "long_running"))` and it'll work the same since (or (null)) evals to (false)

Reply:
I see, gonna keep it this way for the pattern/readability though